lib/metasploit/framework/login_scanner/gitlab.rb

```diff
@@ -0,0 +1,96 @@
+require 'metasploit/framework/login_scanner/http'
+
+module Metasploit
+  module Framework
+    module LoginScanner
+      # GitLab login scanner
+      class GitLab < HTTP
+        # Inherit LIKELY_PORTS,LIKELY_SERVICE_NAMES, and REALM_KEY from HTTP
+        CAN_GET_SESSION = false
+        DEFAULT_PORT    = 80
+        PRIVATE_TYPES   = [ :password ]
+
+        # (see Base#set_sane_defaults)
+        def set_sane_defaults
+          self.uri = '/users/sign_in' if uri.nil?
+          self.method = 'POST' if method.nil?
+
+          super
+        end
+
+        def attempt_login(credential)
+          result_opts = {
+            credential: credential,
+            host: host,
+            port: port,
+            protocol: 'tcp',
+            service_name: ssl ? 'https' : 'http'
+          }
+          begin
+            cli = Rex::Proto::Http::Client.new(host,
+                                               port,
+                                               {
+                                                 'Msf' => framework,
+                                                 'MsfExploit' => framework_module
+                                               },
+                                               ssl,
```

Comment:
@Meatballs1 Can you please test this option right quick and see if it works? I'm currently working on a login module myself, and I have to explicitly add a :ssl and :ssl_version accessor in my LoginScanner class in order to set them.